physical/file.go

```diff
@@ -22,33 +22,62 @@ import (
 // and non-performant. It is meant mostly for local testing and development.
 // It can be improved in the future.
 type FileBackend struct {
-	Path   string
-	l      sync.Mutex
-	logger log.Logger
+	sync.RWMutex
+	path       string
+	logger     log.Logger
+	permitPool *PermitPool
 }
 
-// newFileBackend constructs a Filebackend using the given directory
+type TransactionalFileBackend struct {
+	FileBackend
+}
+
+// newFileBackend constructs a FileBackend using the given directory
 func newFileBackend(conf map[string]string, logger log.Logger) (Backend, error) {
 	path, ok := conf["path"]
 	if !ok {
 		return nil, fmt.Errorf("'path' must be set")
 	}
 
 	return &FileBackend{
-		Path:   path,
-		logger: logger,
+		path:       path,
+		logger:     logger,
+		permitPool: NewPermitPool(DefaultParallelOperations),
+	}, nil
+}
+
+func newTransactionalFileBackend(conf map[string]string, logger log.Logger) (Backend, error) {
+	path, ok := conf["path"]
```

Comment:
[Optional] We could choose to call newFileBackend and set the returned FileBackend in TransactionalFileBackend.